tools/const_finder/test/const_finder_test.dart

```diff
@@ -165,71 +197,248 @@ void _checkNonConsts() {
+      compiler: Compiler.aot,
+    ),
+    _Test(
+      name: 'consts_and_non_web',
+      dartSource: path.join(fixtures, 'lib', 'consts_and_non.dart'),
+      frontendServer: frontendServer,
+      sdkRoot: sdkRoot,
+      librariesSpec: librariesSpec,
+      verify: _checkNonConstsWeb,
+      compiler: Compiler.dart2js,
+    ),
+    _Test(
+      name: 'static_icon_provider_frontend',
+      dartSource: path.join(fixtures, 'lib', 'static_icon_provider.dart'),
+      frontendServer: frontendServer,
+      sdkRoot: sdkRoot,
+      librariesSpec: librariesSpec,
+      verify: _checkAnnotation,
+      compiler: Compiler.aot,
+    ),
+    _Test(
+      name: 'static_icon_provider_web',
+      dartSource: path.join(fixtures, 'lib', 'static_icon_provider.dart'),
+      frontendServer: frontendServer,
+      sdkRoot: sdkRoot,
+      librariesSpec: librariesSpec,
+      verify: _checkAnnotation,
+      compiler: Compiler.dart2js,
+    ),
+  ];
   try {
-    void checkProcessResult(ProcessResult result) {
-      if (result.exitCode != 0) {
-        stdout.writeln(result.stdout);
-        stderr.writeln(result.stderr);
+    stdout.writeln('Generating kernel fixtures...');
+
+    for (final _Test test in tests) {
+      test.run();
+    }
+  } finally {
+    try {
+      for (final _Test test in tests) {
+        test.dispose();
       }
-      expect(result.exitCode, 0);
+    } finally {
+      stdout.writeln('Tests ${exitCode == 0 ? 'succeeded' : 'failed'} - exit code: $exitCode');
     }
+  }
+}
 
-    stdout.writeln('Generating kernel fixtures...');
-    stdout.writeln(consts);
+enum Compiler {
+  // Uses TFA tree-shaking.
+  aot,
+  // Does not have TFA tree-shaking.
+  dart2js,
+}
 
-    checkProcessResult(Process.runSync(dart, <String>[
-      frontendServer,
-      '--sdk-root=$sdkRoot',
-      '--target=flutter',
-      '--aot',
-      '--tfa',
-      '--packages=$packageConfig',
-      '--output-dill=$boxDill',
-      box,
-    ]));
+class _Test {
+  _Test({
+    required this.name,
+    required this.dartSource,
+    required this.sdkRoot,
+    required this.verify,
+    required this.frontendServer,
+    required this.librariesSpec,
+    required this.compiler,
+  }) : dillPath = path.join(fixtures, '$name.dill');
+
+  final String name;
+  final String dartSource;
+  final String sdkRoot;
+  final String frontendServer;
+  final String librariesSpec;
+  final String dillPath;
+  void Function(String, Compiler) verify;
+  final Compiler compiler;
 
+  final List<String> resourcesToDispose = <String>[];
+
+  void run() {
+    stdout.writeln('Compiling $dartSource to $dillPath with $compiler');
+
+    if (compiler == Compiler.aot) {
+      _compileAOTDill();
+    } else {
+      _compileWebDill();
+    }
+
+    stdout.writeln('Testing $dillPath');
+
+    verify(dillPath, compiler);
+  }
+
+  void dispose() {
+    for (final String resource in resourcesToDispose) {
+      stdout.writeln('Deleting $resource');
+      File(resource).deleteSync();
```

Comment:
This line needs to be wrapped in an exception handler.  The failure to find and delete this file in the cleanup phase is causing the test to fail, and our HHH builders to fail (testing head of flutter/flutter, flutter/engine, and dart-lang/sdk together).

Reply:
This may have been failing because of a change that tried to use the prebuilt SDK, which was reverted.  Could it be interacting with https://github.com/flutter/flutter/pull/115242?

The hhh and monorepo builders should be able to build the full SDK, or whatever is needed.  But they do not have a prebuilt Dart SDK that they download.  The monorepo builder and HHH builder do use --local-engine when running flutter commands: https://dart.googlesource.com/monorepo/+/refs/heads/main

Reply:
I don't really have context into this issue. I reverted https://github.com/flutter/flutter/pull/115242 because it broke the flutter tool for me on an m1. I can't imagine that any recepie changes depended on that yet, because the build would have broken after I reverted, or before I landed.

Reply:
@whesse I think the action item here is--when building the Dart SDK on the HHH bot, ensure that  //engine/src/out/host_debug_unopt/dart-sdk/bin/snapshots/dart2js.dart.snapshot is built (I'm assuming that //engine/src/out/host_debug_unopt/dart-sdk/bin/dart is present, since this script also references that binary).

Reply:
This has caused our HHH builds to fail. The modes where we run `const_finder.dart` need to have full dart sdk built, since it uses now dart2js. So the GN invocation needs `--full-dart-sdk`.

On Dart side, I'll updat our recipes in [recipes/cl/271382](https://dart-review.googlesource.com/c/recipes/+/271382) and [monorepo/cl/271384](https://dart-review.googlesource.com/c/monorepo/+/271384)